modules/metrics/src/main/java/com/spotify/apollo/metrics/semantic/SemanticRequestMetrics.java

```diff
@@ -24,35 +24,60 @@
 import com.codahale.metrics.Histogram;
 import com.codahale.metrics.Meter;
 import com.codahale.metrics.RatioGauge;
+import com.codahale.metrics.RatioGauge.Ratio;
+import com.codahale.metrics.Timer;
+import com.spotify.apollo.Request;
+import com.spotify.apollo.Response;
 import com.spotify.apollo.StatusType;
 import com.spotify.apollo.metrics.RequestMetrics;
-import com.spotify.apollo.metrics.TimerContext;
 import com.spotify.metrics.core.MetricId;
 import com.spotify.metrics.core.SemanticMetricRegistry;
 
+import java.util.Optional;
+import java.util.Set;
 import java.util.function.Supplier;
 
+import okio.ByteString;
+
 import static com.spotify.apollo.StatusType.Family.INFORMATIONAL;
 import static com.spotify.apollo.StatusType.Family.SUCCESSFUL;
+import static com.spotify.apollo.metrics.semantic.Metric.DROPPED_REQUEST_RATE;
+import static com.spotify.apollo.metrics.semantic.Metric.ENDPOINT_REQUEST_DURATION;
+import static com.spotify.apollo.metrics.semantic.Metric.ENDPOINT_REQUEST_RATE;
+import static com.spotify.apollo.metrics.semantic.Metric.ERROR_RATIO;
+import static com.spotify.apollo.metrics.semantic.Metric.REPLY_SIZE;
+import static com.spotify.apollo.metrics.semantic.Metric.REQUEST_FANOUT_FACTOR;
+import static com.spotify.apollo.metrics.semantic.Metric.REQUEST_SIZE;
 import static java.util.Objects.requireNonNull;
 
+// Optional fields are fine; they enable the use of the 'ifPresent' idiom which is more readable
+// than if statements
+@SuppressWarnings("OptionalUsedAsFieldOrParameterType")
 class SemanticRequestMetrics implements RequestMetrics {
 
   private static final String COMPONENT = "service-request";
-  private final SemanticMetricRegistry metricRegistry;
 
-  private final MetricId countRequestId;
-  private final MetricId timeRequestId;
+  private final SemanticMetricRegistry metricRegistry;
+  private final Set<Metric> enabledMetrics;
   private final Meter sentReplies;
   private final Meter sentErrors;
-  private final Histogram fanoutHistogram;
+
+  private final MetricId requestRateId;
+  private final Optional<Histogram> fanoutHistogram;
+  private final Optional<Histogram> requestSizeHistogram;
+  private final Optional<Histogram> replySizeHistogram;
+  private final Optional<Timer.Context> timerContext;
+  private final Optional<Meter> droppedRequests;
 
   SemanticRequestMetrics(
```

Comment:
Move much of the things done in the SemanticRequestMetrics constructor to SemanticServiceMetrics constructor?